### DISCUSSION TITLE
Caddy configuration for Jellyfin

### DISCUSSION THREAD

**Discussion by robocrax** [hrfee/jfa-go#133] [Show and tell]
For the wiki: [Password resets outside local network](https://wiki.jfa-go.com/docs/password-resets/)

I needed to created the same configuration for Caddy and took me ages to find it (because I'm very proficient in reading simple docs) so could you please add this below the NGINX config so that next I stumble upon this I know what to do.

```
<jellyfin address> {

    # rest of jellyfin config

    reverse_proxy /Users/ForgotPassword localhost:8096,
    reverse_proxy /Users/ForgotPassword/Pin localhost:8096 {
        header_up X-Forwarded-For <any local ip address>
    }

    reverse_proxy localhost:8096

}
```

**Comment by hrfee** [comment]
Great, i'll add it now. Thanks!

**Comment by hrfee** [comment]
By the way, the `/Users/ForgotPassword` thing is meant for Jellyfin itself, not jfa-go so it isn't necessary.

**Reply by robocrax** [reply]
> By the way, the `/Users/ForgotPassword` thing is meant for Jellyfin itself, not jfa-go so it isn't necessary.

Hey hrfee, yes I am configuring Jellyfin and not jfa-go directly but it is kinda required since I can't request a reset in the first place.

I just copied the existing paths from the NGINX config and it works on my setup.

**Comment by FaNt4zMa** [comment]
@robocrax does still work with the latest JF version? I tried adding these lines to my caddy config but Jellyfin still sees my external IP and refuses to initiate the password reset. According to the API docs, the path are still valid.. Any idea?
Thanks

Edit: Im starting to think its due to my public domain being duckdns...? I was doing some troubleshooting with ChatGPT and it seems like my configuration is right but running a `Invoke-RestMethod -Uri https://[redacted].duckdns.org/Users/ForgotPassword -Method Post -Headers @{ "Content-Type" = "application/json"; "X-Forwarded-For" = "192.168.2.55" } -Body '{"EnteredUsername": "Viewer"}'` command returns `InNetworkRequired`, which indicates that the /User/ForgotPassword endpoint also needs the header X-Forwarded-For. Maybe the [example](https://wiki.jfa-go.com/docs/pwr/remote-network/) should be modified.

Edit2: I figured adding my Caddy IP and DuckDNS domain to the list of known proxies for Jellyfin can't hurt as it says: This is required to make proper use of 'X-Forwarded-For' headers. We'll see...

Edit3: Nevermind, I made it work somehow! My current caddy config looks like [this](https://pastebin.com/RK19UUhe) and I have both my public domain and caddy IP in my Known Proxies settings in Jellyfin. One of these things made it work, if that can help anyone else that has a DuckDNS setup 😂
